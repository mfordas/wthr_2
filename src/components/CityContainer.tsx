import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

import {types} from 'app/constants';

interface Props {
  city: types.apiGeocodeCityResponse;
  onPress: () => void;
}

export const CityContainer: React.FC<Props> = ({city, onPress}) => {
  const {name, country, state} = city;
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Text>{name} </Text>
      <Text>{country} </Text>
      {state && <Text>{state}</Text>}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    height: 48,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
