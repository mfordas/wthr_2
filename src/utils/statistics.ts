import {max, min, mean, mode, round} from 'mathjs';

import {types} from 'app/constants';

export const prepareStatisticsData = (
  apiData: types.apiOneDayForecastResponse[],
) => {
  const maxTemperatures = apiData.map(day => round(day.temp.max));
  const minTemperatures = apiData.map(day => round(day.temp.min));

  return {maxTemperatures, minTemperatures};
};

export const countMaxValue = (arrayOfValues: number[]) =>
  round(max(arrayOfValues));

export const countMinValue = (arrayOfValues: number[]) =>
  round(min(arrayOfValues));

export const countModeValue = (arrayOfValues: number[]) => {
  const modeOfNumbers: number[] = mode(arrayOfValues);
  return modeOfNumbers.length === arrayOfValues.length ? false : modeOfNumbers;
};

export const countMeanValue = (arrayOfValues: number[]) =>
  round(mean(arrayOfValues));
