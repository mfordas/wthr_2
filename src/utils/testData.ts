import {types} from 'app/constants';

export const mockedGeolocationApiCityReponse: types.apiGeocodeCityResponse = {
  name: 'London',
  lat: 12.123456789,
  lon: 12.123456789,
  local_names: [
    {
      af: 'Londen',
    },
  ],
  country: 'GB',
};

export const mockedOneDayForecastApiResponse: types.apiOneDayForecastResponse =
  {
    clouds: 52,
    dew_point: 12.27,
    dt: 1655373600,
    feels_like: {
      day: 24.53,
      eve: 18.1,
      morn: 11.91,
      night: 13.84,
    },
    humidity: 46,
    moon_phase: 0.58,
    moonrise: 1655415960,
    moonset: 1655352300,
    pop: 0.98,
    pressure: 1017,
    rain: 3.43,
    sunrise: 1655346977,
    sunset: 1655406530,
    temp: {
      day: 24.79,
      eve: 18,
      max: 26.05,
      min: 12.47,
      morn: 12.47,
      night: 14.13,
    },
    uvi: 5.09,
    weather: [
      {
        description: 'moderate rain',
        icon: '10d',
        id: 501,
        main: 'Rain',
      },
    ],
    wind_deg: 310,
    wind_speed: 6.65,
  };
