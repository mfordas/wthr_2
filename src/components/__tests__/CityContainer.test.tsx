import {render} from '@testing-library/react-native';
import React from 'react';

import {mockedGeolocationApiCityReponse} from 'app/utils/testData';

import {CityContainer, Props} from '../CityContainer';

const props: Props = {
  city: mockedGeolocationApiCityReponse,
  onPress: jest.fn(),
};
const wrapper = render(<CityContainer {...props} />);
test('CityContainer should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});
