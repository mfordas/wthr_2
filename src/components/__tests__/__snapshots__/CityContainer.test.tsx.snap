// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityContainer should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text>
    London
     
  </Text>
  <Text>
    GB
     
  </Text>
</View>
`;
